src/CMakeLists.txt

```diff
@@ -0,0 +1,130 @@
+# # ------------------------------------------------------------------------- #
+# # ufs-da-bundle for building all the components needed for the UFS-DA system #
+# # ------------------------------------------------------------------------- #
+
+
+# Preamble
+# --------
+
+# Check for minimim cmake requirement
+cmake_minimum_required( VERSION 3.12 FATAL_ERROR )
+
+# Create project
+project( ufsda-bundle VERSION 1.0.0 LANGUAGES C CXX Fortran )
+
+# Include ecbuild_bundle macro
+include( ecbuild_bundle )
+
+# Default build mode, change with --build=[release]|debug|bit
+set( ECBUILD_DEFAULT_BUILD_TYPE Release )
+
+# Enable MPI
+set( ENABLE_MPI ON CACHE BOOL "Compile with MPI" )
+
+
+# Initialize bundle
+# -----------------
+ecbuild_bundle_initialize()
+
+
+# Add packages to the bundle
+# --------------------------
+
+# jedi-cmake
+ecbuild_bundle( PROJECT jedicmake GIT "https://github.com/jcsda/jedi-cmake.git" BRANCH develop UPDATE )
+
+# ECMWF libraries
+option("BUNDLE_SKIP_ECKIT" "Don't build eckit" "ON" ) # Skip eckit build unless user passes -DBUNDLE_SKIP_ECKIT=OFF
+option("BUNDLE_SKIP_FCKIT" "Don't build fckit" "ON") # Skip fckit build unless user passes -DBUNDLE_SKIP_FCKIT=OFF
+option("BUNDLE_SKIP_ATLAS" "Don't build atlas" "ON") # Skip atlas build unless user passes -DBUNDLE_SKIP_ATLAS=OFF
+
+# turn off optional OOPS toy models
+option( ENABLE_LORENZ95_MODEL "Build LORENZ95 toy model" OFF )
+option( ENABLE_QG_MODEL "Build QG toy model" OFF )
+
+ecbuild_bundle( PROJECT eckit GIT "https://github.com/ecmwf/eckit.git" TAG 1.16.0 )
+ecbuild_bundle( PROJECT fckit GIT "https://github.com/ecmwf/fckit.git" TAG 0.9.2 )
+ecbuild_bundle( PROJECT atlas GIT "https://github.com/ecmwf/atlas.git" TAG 0.24.1 )
+
+# External (required) observation operators
+ecbuild_bundle( PROJECT crtm GIT "https://github.com/jcsda/crtm.git"   TAG v2.3-jedi.3 )
+
+# Core JEDI repositories
+ecbuild_bundle( PROJECT oops  GIT "https://github.com/jcsda/oops.git"  BRANCH develop UPDATE )
+ecbuild_bundle( PROJECT saber GIT "https://github.com/jcsda/saber.git" BRANCH develop UPDATE )
+ecbuild_bundle( PROJECT ioda  GIT "https://github.com/jcsda/ioda.git"  BRANCH develop UPDATE )
+ecbuild_bundle( PROJECT ufo   GIT "https://github.com/noaa-emc/ufo.git"   BRANCH feature/ufsda UPDATE )
+
+# FMS and FV3 dynamical core
+ecbuild_bundle( PROJECT fms GIT "https://github.com/jcsda/FMS.git" BRANCH release-stable UPDATE )
+ecbuild_bundle( PROJECT fv3 GIT "https://github.com/jcsda/GFDL_atmos_cubed_sphere.git" BRANCH release-stable UPDATE )
+
+# fv3-jedi and associated repositories
+ecbuild_bundle( PROJECT femps       GIT "https://github.com/jcsda/femps.git"                BRANCH develop UPDATE )
+ecbuild_bundle( PROJECT fv3-jedi-lm GIT "https://github.com/jcsda/fv3-jedi-linearmodel.git" BRANCH develop UPDATE )
+ecbuild_bundle( PROJECT fv3-jedi    GIT "https://github.com/jcsda/fv3-jedi.git"             BRANCH develop UPDATE )
+
+# SOCA associated repositories
```

Comment:
:tada: :tada: :tada:

Reply:
soca is NOT an option! I like it :) but it might not be very practical for the short term. Something we can revisit in the near future.